src/Coordination/KeeperStorage.cpp

```diff
@@ -1099,26 +1611,22 @@ KeeperStorage::ResponsesForSessions KeeperStorage::processRequest(const Coordina
 
     if (zk_request->getOpNum() == Coordination::OpNum::Close) /// Close request is special
     {
-        auto it = ephemerals.find(session_id);
-        if (it != ephemerals.end())
-        {
-            for (const auto & ephemeral_path : it->second)
-            {
-                container.updateValue(parentPath(ephemeral_path), [&ephemeral_path] (KeeperStorage::Node & parent)
-                {
-                    --parent.stat.numChildren;
-                    ++parent.stat.cversion;
-                    auto base_name = getBaseName(ephemeral_path);
-                    parent.removeChild(base_name);
-                });
+        commit(zxid, session_id);
 
-                container.erase(ephemeral_path);
+        for (const auto & delta : uncommitted_state.deltas)
```

Comment:
We need to remove all ephemerals for the closed session. Why do we process deltas here? Maybe add a comment?

Reply:
We cannot remove until we commit, we only saw the Close request and we prepared everything for commit, which is `RemoveNodeDelta` for all ephemerals which will be removed on commit.